shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1020,6 +1065,12 @@ - (void)dispatchTouches:(NSSet*)touches
     }
 
     packet->SetPointerData(pointer_index++, pointer_data);
+
+    if (touch.phase == UITouchPhaseEnded || touch.phase == UITouchPhaseCancelled) {
```

Comment:
does this need to check flutter_state_is_added?

Reply:
Engine shell does not need to synthesize. The event will be `kAdd` as needed based on state of the gesture recognizer (see `generatePointerDataForMouse`). Use of `flutter_state_is_added` is only to synchronize between hover recognizer and scroll recognizer. 

flutter_state_is_added is only for the mouse cursor (hovering only). The quoted code is for each UITouch (it's because UIKit uses touch mechanism to send mouse clicks).